api/client/src/main/java/org/projectnessie/client/http/HttpClientBuilderImpl.java

```diff
@@ -279,7 +288,25 @@ public HttpClient build() {
       throw new IllegalArgumentException(
           "No HTTP client factory for name '" + clientName + "' found");
     }
-    return httpClientFactory.buildClient(config);
+    HttpClient client = httpClientFactory.buildClient(config);
+    if (cancellationCallbackConsumer != null) {
+      cancellationCallbackConsumer.accept(client::close);
+    }
+
+    if (authentication != null) {
+      try {
+        authentication.start();
+      } catch (RuntimeException e) {
```

Comment:
Is this catch a hypothetical situation, or are you seeing actual errors here with your experiments?

Asking because `start()` currently cannot throw (and cannot block either): even with OAuth2, the only thing it does is that it completes the `started` future. The actual token fetching happens asynchronously. If an error occurs during token fetching, it will surface to the caller only when calling `OAuth2Authenticator.authenticate()` at the first HTTP request – not here.

Reply:
It is only a safety net